src/widgets/QDisassemblyView.cpp

```diff
@@ -1370,7 +1370,8 @@ void QDisassemblyView::drawJumpArrows(QPainter &painter, const DrawingContext *c
 		// if direct jmp is selected, then draw arrow in red
 		if (is_unconditional_jump(instructions_[jump_arrow.src_line]) && 
 			(ctx->selected_line == jump_arrow.src_line || 
-			(ctx->selected_line == jump_arrow.dst_line && show_addresses_[jump_arrow.dst_line] != current_address_ ))) {
+			(ctx->selected_line == jump_arrow.dst_line && 
+			(jump_arrow.dst_in_viewport && show_addresses_[jump_arrow.dst_line] != current_address_ )))) {
```

Comment:
Nitpick: the additional parentheses are superfluous here, and they do add to clutter on the closing side: `))))`. I'd suggest removing this pair, retaining the logically related conditions on the single line as they are now.

Reply:
Fixed